kong/utils.go

```diff
@@ -452,12 +452,27 @@ func fillConfigRecord(schema gjson.Result, config Configuration, opts FillRecord
 		// the defaults for deprecated fields from the already formed default config
 		backwardTranslation := value.Get(fname + ".translate_backwards")
 
-		if !backwardTranslation.Exists() {
+		var replacePath gjson.Result
+		replacedWith := value.Get(fname + ".deprecation.replaced_with")
+		if replacedWith.IsArray() {
+			for _, item := range replacedWith.Array() {
+				if pathArray := item.Get("path"); pathArray.Exists() && pathArray.IsArray() {
+					replacePath = pathArray
+					break
+				}
+			}
+		}
```

Comment:
looks like we don't need to run any of this code if `backwardTranslation.Exists()` so we could execute this conditionally to `backwardTranslation` not existing.

I think we may also simplify and just reassign `backwardTranslation = pathArray` here if the item is found, so we wouldn't need the `replacePath` variable.